SignalMessaging/attachments/SharingThreadPickerViewController.m

```diff
@@ -233,44 +216,183 @@ - (void)tryToSendAttachment:(SignalAttachment *)attachment fromViewController:(U
                                    completion:^(void) {
-                                       TSOutgoingMessage *outgoingMessage =
+                                       __block TSOutgoingMessage *outgoingMessage =
                                            [ThreadUtil sendMessageWithAttachment:self.attachment
                                                                         inThread:self.thread
                                                                    messageSender:self.messageSender
-                                                                      completion:sendCompletion];
+                                                                      completion:^(NSError *_Nullable error) {
+                                                                          sendCompletion(error, outgoingMessage);
+                                                                      }];
 
                                        self.outgoingMessage = outgoingMessage;
                                    }];
 }
 
 - (void)showSendFailureAlertWithError:(NSError *)error
-                           attachment:(SignalAttachment *)attachment
+                              message:(TSOutgoingMessage *)message
                    fromViewController:(UIViewController *)fromViewController
 {
     AssertIsOnMainThread();
+    OWSAssert(error);
+    OWSAssert(message);
+    OWSAssert(fromViewController);
 
     NSString *failureTitle = NSLocalizedString(@"SHARE_EXTENSION_SENDING_FAILURE_TITLE", @"Alert title");
 
-    UIAlertController *failureAlert = [UIAlertController alertControllerWithTitle:failureTitle
-                                                                          message:error.localizedDescription
-                                                                   preferredStyle:UIAlertControllerStyleAlert];
-
-    UIAlertAction *failureCancelAction = [UIAlertAction actionWithTitle:[CommonStrings cancelButton]
-                                                                  style:UIAlertActionStyleCancel
-                                                                handler:^(UIAlertAction *_Nonnull action) {
-                                                                    [self.shareViewDelegate shareViewWasCancelled];
-                                                                }];
-    [failureAlert addAction:failureCancelAction];
-
-    UIAlertAction *retryAction =
-        [UIAlertAction actionWithTitle:[CommonStrings retryButton]
-                                 style:UIAlertActionStyleDefault
-                               handler:^(UIAlertAction *action) {
-                                   [self tryToSendAttachment:attachment fromViewController:fromViewController];
-                               }];
-    [failureAlert addAction:retryAction];
-
-    [fromViewController presentViewController:failureAlert animated:YES completion:nil];
+    if (error.code == OWSErrorCodeUntrustedIdentity) {
```

Comment:
Don't you need to check the error domain too?

Reply:
Done!